src/main/java/com/openlattice/shuttle/EntityDefinition.java

```diff
@@ -47,44 +46,46 @@
 public class EntityDefinition implements Serializable {
 
     private static final long serialVersionUID = -3565689091187367622L;
 
-    private static final Logger logger = LoggerFactory.getLogger( EntityDefinition.class );
-
-    private final FullQualifiedName                                           entityTypeFqn;
-    private final String                                                      entitySetName;
-    private final List<FullQualifiedName>                                     key;
-    private final Map<FullQualifiedName, PropertyDefinition>                  propertyDefinitions;
-    private final String                                                      alias;
-    public final  Optional<Conditions>                                        condition;
-    private final Optional<SerializableFunction<Map<String, String>, String>> generator;
-    private final boolean                                                     useCurrentSync;
-    public final  SerializableFunction<Map<String, String>, ?>                valueMapper;
+    private static final Logger                                                      logger = LoggerFactory
+            .getLogger( EntityDefinition.class );
+    protected final      Optional<Conditions>                                        condition;
+    protected final      SerializableFunction<Map<String, Object>, ?>                valueMapper;
+    protected final      Optional<UUID>                                              id;
+    protected final      FullQualifiedName                                           entityTypeFqn;
+    protected final      String                                                      entitySetName;
+    protected final      List<FullQualifiedName>                                     key;
+    protected final      Map<FullQualifiedName, PropertyDefinition>                  propertyDefinitions;
+    protected final      String                                                      alias;
+    protected final      Optional<SerializableFunction<Map<String, Object>, String>> generator;
+    protected final      UpdateType                                                  updateType;
 
     @JsonCreator
     public EntityDefinition(
+            @JsonProperty( SerializationConstants.ID_FIELD ) Optional<UUID> id,
             @JsonProperty( SerializationConstants.FQN ) String entityTypeFqn,
             @JsonProperty( SerializationConstants.ENTITY_SET_NAME ) String entitySetName,
             @JsonProperty( SerializationConstants.KEY_FIELD ) List<FullQualifiedName> key,
             @JsonProperty( SerializationConstants.PROPERTY_DEFINITIONS )
                     Map<FullQualifiedName, PropertyDefinition> propertyDefinitions,
             @JsonProperty( SerializationConstants.NAME ) String alias,
             @JsonProperty( Constants.CONDITIONS ) Optional<Conditions> condition,
-            @JsonProperty( SerializationConstants.CURRENT_SYNC ) Boolean useCurrentSync ) {
```

Comment:
Removing useCurrentSync will break all the existing integrations, so we should update all of those before merging this.

Reply:
I don't think this is still working, and all integrations are in fact CurrentSyncs...